### PR TITLE
Broadcast connection-related events

### PR DESCRIPTION
**Pull request by rhauch**
## Summary of Changes

Add a notional WebSocket message type for signaling connection-related changes, such as when connections are created, updated, deleted, or when their status changes.

## Pull request checklist

Please check if your PR fulfills the following (if applicable):

- Tests:
    - [x] Added new
    - [x] Updated existing
    - [ ] Deleted existing
- [ ] Have you validated this change locally against a running instance of the Quarkus dev server?
    ```shell
    make quarkus-dev
    ```
- [ ] Have you validated this change against a locally running native executable?
    ```shell
    make mvn-package-native && ./target/ide-sidecar-*-runner
    ```

### CODE REVIEW

**Review by jlrobins — APPROVED** [review]
Looks great!

**Comment by rhauch** [comment]
Still needs a test to verify the events are being fired when connections change. Marking as draft...

File: src/main/java/io/confluent/idesidecar/websocket/resources/ConnectionEvents.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  }
+
+  void broadcast(ConnectionState connection, Action action) {
+    broadcast(
+        new ConnectionEventBody(connection, action),
+        new MessageHeaders(MessageType.CONNECTION_EVENT)
+    );
+  }
+
+  void broadcast(ConnectionEventBody body, MessageHeaders headers) {
```
Comment (jlrobins) [inline src/main/java/io/confluent/idesidecar/websocket/resources/ConnectionEvents.java:53]: `(body, headers)` What is this, reverse Polish notation?

**Comment by rhauch** [comment]
> Still needs a test to verify the events are being fired when connections change. Marking as draft...

Later commits add unit tests to verify the events are being fired, albeit without real connections. Changing from draft to be ready for review.

File: src/test/java/io/confluent/idesidecar/websocket/resources/ConnectionEventsTest.java [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+      "http://localhost:1234/gateway/v1/connections/" + SPEC.id(),
+      null,
+      null
+  );
+
+  @Inject
+  ConnectionEvents connectionEvents;
+
+  @Test
+  void shouldBroadcastConnectionEvents() {
```
Comment (jlrobins) [inline src/test/java/io/confluent/idesidecar/websocket/resources/ConnectionEventsTest.java:64]: Love it!

**Review by jlrobins — DISMISSED** [review]

**Review by jlrobins — APPROVED** [review]

**Pull request merged**